### DISCUSSION TITLE
Mutating the FCM message

### DISCUSSION THREAD

**Discussion by amine-louni** [invertase/react-native-firebase#5675] [Q&A] [answered]
Is there any solution to mutate the remoteMessage message before firing up the push notification ?

**Comment by mikehardy** [comment, marked as answer]
Hi there! You are not providing many details about your use case, but I think you want something like a notification service extension, e.g. https://rnfirebase.io/messaging/ios-notification-images or https://notifee.app/react-native/docs/ios/remote-notification-support
